executionError_test.go

```diff
@@ -0,0 +1,43 @@
+package gantry
+
+import (
+	"fmt"
+	"os/exec"
+	"testing"
+)
+
+func TestExecutionErrorError(t *testing.T) {
+	msg := "I am an Error"
+	e := ExecutionError{
+		err:      fmt.Errorf(msg),
+		override: 1,
+	}
+	if e.Error() != msg {
+		t.Errorf("incorrect error message, got: %s wanted: %s", e.Error(), msg)
+	}
+}
+
+func TestExecutionErrorExitCodeSimpleOverride(t *testing.T) {
+	msg := "I am an Error"
+	e := ExecutionError{
+		err:      fmt.Errorf(msg),
+		override: 0,
```

Comment:
Why is this called `override` and not `exitCode` like the method that accesses it?

Reply:
Currently the only place where this error is generated is:
https://github.com/ad-freiburg/gantry/blob/7717085f3547a38a6f242ddc4b367f9429f2f4ac/pipeline.go#L396
Here `err` is always an `exec.ExitError` which already has an `ExitCode()` which is returned, if `override == 0`. If it's not `0` it will be returned instead thus overriding the real error code.
As a fallback `1` is returned if no override is set and somehow the inner `err` is not a `exec.ExitError`:
https://github.com/ad-freiburg/gantry/blob/0b1d1513a54f7e8f9455bca55f8b3932c997ca00/executionError.go#L19-L27

The name is not set in stone but I think naming it `execCode` would not convey the correct implications of setting a value.

Reply:
How about `exitCodeOverride`?